### DISCUSSION TITLE
pwa bookmark icons on ipad homescreen

### DISCUSSION THREAD

**Discussion by zhaob1n** [coder/code-server#4983] [Q&A] [answered]
I installed code-server 4.1.0 today and I tried to add the website to the home screen of my ipad. The thing is, if I add the login webpage, pwa failed, and if I add the webpage after loging in, the icon shows the webview which I don't like. Actually months there is no such problem, adding the login page won't fail the pwa. Is there a soluton?
![IMG_0133(20220311-165732)](https://user-images.githubusercontent.com/52552971/157841422-e601eb37-cb56-41d9-a10e-fb179cd1762b.PNG)
![IMG_0134(20220311-165843)](https://user-images.githubusercontent.com/52552971/157841432-7dcc3265-9328-4f65-ad85-0185fdbddd47.PNG)

**Comment by lg-nineoneninethreenine** [comment]
You can actually edit the name when setting the PWA. I think it's possible on Safari but if Apple closed that, then you can try Google Chrome.

**Reply by zhaob1n** [reply]
I mean the icon. Can I change the icon to the first picture with pwa still enabled.

**Reply by lg-nineoneninethreenine** [reply]
Hmm I don't know, never bought an iPad in my life... @jsjoeio can you tell how to change PWA icons or at lease change some setting?

**Reply by lg-nineoneninethreenine** [reply]
Wait, that's it! A custom iPadOS setting related to Safari or Chrome that shows webpage previews than the favicon!!! Check for anything of that sort now!

**Reply by jsjoeio** [reply]
Hmm...I don't know if you can change the icon. I did a quick Google search but I'm not finding anything.

**Reply by zhaob1n** [reply]
> 

I don’t find any setting about that…..

**Comment by mcmans** [comment, marked as answer]
Hi,

You can set a custom icon by following this:

https://www.reddit.com/r/iphone/comments/oa4yq9/set_apple_touch_icon_for_any_website/

I’ve done it to set it to the VSCode Icon

**Comment by lg-nineoneninethreenine** [comment]
I think you could use the Shortcuts app and make it in such a way that it opens a website rather than the app. How's my idea? Got this from the WWW.

**Comment by umdstu** [comment]
This definitely seems like a bug. It must have something to do with the meta tags. The login page header differs completely from the logged in page header. I'm going to try to play around with what might make it work, but an external workaround shouldn't be the solution.

**Reply by jsjoeio** [reply]
If you find out it is indeed a bug, feel free to open a bug report!

**Reply by umdstu** [reply]
I did find the issue. I'll create a separate ticket sometime today.

**Reply by jsjoeio** [reply]
Thanks so much!

**Reply by umdstu** [reply]
https://github.com/coder/code-server/issues/4994
